Realm/Realm/Logging/Logger.cs

```diff
@@ -118,27 +152,29 @@ protected Logger()
             _gcHandle = new Lazy<GCHandle>(() => GCHandle.Alloc(this));
         }
 
-        internal static void LogDefault(LogLevel level, string message) => Default?.Log(level, message);
+        internal static void LogDefault(LogLevel level, string message, LogCategory? category = null) => Default?.Log(level, message, category);
 
         /// <summary>
-        /// Log a message at the supplied level.
+        /// Log a message at the supplied level and category.
         /// </summary>
         /// <param name="level">The criticality level for the message.</param>
         /// <param name="message">The message to log.</param>
-        public void Log(LogLevel level, string message)
+        /// <param name="category">The category for the message. Defaults to <see cref="LogCategory.RealmLogCategory.SDK"/> if not specified.</param>
+        public void Log(LogLevel level, string message, LogCategory? category = null)
```

Comment:
Have you explored overloading this fuction instead of modifying the signature to keep compatibility?

We could have two functions one with the old signature, and the new one. It would be easier later on to deprecate the old behavior.

Reply:
This started off as:

```csharp
public void Log(LogLevel level, string message);
public void Log(LogLevel level, LogCategory category, string message);
```

But we updated it to have a default last param briefly discussed in [this thread](https://github.com/realm/realm-dotnet/pull/3634#discussion_r1655062184). My current understanding is that we don't want to deprecate `Log(level, message)` but to keep providing the option not to provide an explicit category.

Although if this was to change I agree that it'd be clearer to have overloads. Personally, I think the DX feels better having `level, category, message` over a `level, message, category` alternative if we can provide this w/o breaking APIs (including for `LogImpl()`).

Reply:
Update: Overloads are now being used, and notes on avoiding a breaking change has been added to the PR description.